### PR TITLE
Running mean

### PR DESCRIPTION
**Pull request by dilnoza92**
Updated the eds model files with the new functions (running mean) and updated functions.

### CODE REVIEW

File: tensorflow_plugin/models/example-models/eds/real_build.py [hunk 1]
```diff
@@ -1,58 +1,87 @@
...
+    
+    bias_energy=tf.Variable(tf.ones(N,dtype=tf.float32), dtype=tf.float32,name='energy')
+    steps = tf.Variable(1.0, name='steps')#,dtype=tf.float64)
+    mean = tf.Variable(0.1, name='mean')#,dtype=tf.float64)
+    variance=tf.Variable(0.1, name='variance')#,dtype=tf.float64)
+    gradient=tf.Variable(0.1,name='gradient')#,dtype=tf.float64)
+    alpha=tf.Variable(1.0,name='alpha')#,dtype=tf.float64)
+    bias_val=tf.Variable(1.0,name='bias-val')#,expected_shape=(1),dtype=tf.float64)
+    set_point=tf.Variable(set_point,name='set_point')#,dtype=tf.float64)
+    #sess = tf.Session()
```
Comment (whitead) [inline tensorflow_plugin/models/example-models/eds/real_build.py]: Please remove unused lines.

File: tensorflow_plugin/models/example-models/eds/real_build.py [hunk 2]
```diff
@@ -1,58 +1,87 @@
...
-    return tf.reduce_mean(rs),print_op
+    print_op=tf.print('center_of_mass is',com,output_stream=sys.stdout)
+    print_op2=tf.print('subtract',tf.math.subtract(graph.positions[:,:2],tf.reshape(tf.tile(com,[N]),(N,2))),output_stream=sys.stdout)
+    
+    aaaa=tf.print('tile',tf.tile(com,[N]),output_stream=sys.stdout)
+    aaa=tf.print('com in long term',tf.reshape(tf.tile(com,[N]),(N,2)),output_stream=sys.stdout)
+    rs= graph.safe_norm(tf.math.subtract(graph.positions[:,:2],com))#,tf.reshape(tf.tile(com,[N]),(N,2))),axis=1)
+
+
+    a=tf.print('rs not reduce mean',rs,output_stream=sys.stdout)
```
Comment (whitead) [inline tensorflow_plugin/models/example-models/eds/real_build.py]: Let's remove prints from the code

File: tensorflow_plugin/models/example-models/eds/real_run.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import hoomd, hoomd.md, hoomd.tensorflow_plugin
```
Comment (whitead) [inline tensorflow_plugin/models/example-models/eds/real_run.py]: Can we change the name from `real_run.py` ro just `run.py`?

File: tensorflow_plugin/models/example-models/eds/real_run.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
-
-with hoomd.tensorflow_plugin.tfcompute('/tmp/eds') as tfcompute:
+import tensorflow as tf
+tf.ConfigProto(allow_soft_placement=True, log_device_placement=True)
+#output_aver_r_running_mean.txt
+with hoomd.tensorflow_plugin.tfcompute('manuscrpt/eds') as tfcompute:
```
Comment (whitead) [inline tensorflow_plugin/models/example-models/eds/real_run.py]: Let's be careful with the path names. Maybe just do `'/tmp/eds'`

File: tensorflow_plugin/models/example-models/eds/real_build.py [hunk 5]
```diff
@@ -1,58 +1,87 @@
-import hoomd.tensorflow_plugin
+import hoomd.tensorflow_plugin as htf
+import sys
 import tensorflow as tf
 
+
+out_aver=open('output_aver_r_running_mean.txt', 'a')
```
Comment (whitead) [inline tensorflow_plugin/models/example-models/eds/real_build.py]: See the `running_mean` example from the `README.md`. You should not have an open file. The values are automatically saved in an output file.

File: tensorflow_plugin/models/example-models/eds/real_run.py [hunk 6]
```diff
@@ -1,9 +1,11 @@
 import hoomd, hoomd.md, hoomd.tensorflow_plugin
-
-with hoomd.tensorflow_plugin.tfcompute('/tmp/eds') as tfcompute:
+import tensorflow as tf
+tf.ConfigProto(allow_soft_placement=True, log_device_placement=True)
```
Comment (whitead) [inline tensorflow_plugin/models/example-models/eds/real_run.py]: What does this do? Maybe add a comment.

File: tensorflow_plugin/models/example-models/eds/real_run.py [hunk 7]
```diff
@@ -13,6 +15,11 @@
...
     hoomd.md.integrate.nve(
         group=hoomd.group.all()).randomize_velocities(kT=0.2, seed=42)
+    hoomd.analyze.log(filename='eds.log',
+                      quantities = ['momentum', 'temperature', 'time'],
+                      period=100,overwrite=True)
 
     tfcompute.attach(nlist, r_cut=rcut)
-    hoomd.run(100)
+    #hoomd.dump.gsd(filename="trajectory_eds.gsd", period=10)
+    hoomd.dump.gsd(filename="saveall_running.gsd", overwrite=True, period=100, group=hoomd.group.all(), dynamic=['attribute', 'momentum', 'topology'])
```
Comment (whitead) [inline tensorflow_plugin/models/example-models/eds/real_run.py]: Use single quotes

File: tensorflow_plugin/models/example-models/eds/real_run.py (hunk 3 above)
Comment (whitead) [reply tensorflow_plugin/models/example-models/eds/real_run.py]: and `real_build.py` to `build.py`

File: tensorflow_plugin/models/example-models/eds/real_build.py [hunk 8]
```diff
@@ -1,58 +1,87 @@
...
+    A=tf.constant(0.30,name='A',dtype=tf.float32)
+    
+    bias_energy=tf.Variable(tf.ones(N,dtype=tf.float32), dtype=tf.float32,name='energy')
+    steps = tf.Variable(1.0, name='steps')#,dtype=tf.float64)
+    mean = tf.Variable(0.1, name='mean')#,dtype=tf.float64)
+    variance=tf.Variable(0.1, name='variance')#,dtype=tf.float64)
+    gradient=tf.Variable(0.1,name='gradient')#,dtype=tf.float64)
+    alpha=tf.Variable(1.0,name='alpha')#,dtype=tf.float64)
+    bias_val=tf.Variable(1.0,name='bias-val')#,expected_shape=(1),dtype=tf.float64)
+    set_point=tf.Variable(set_point,name='set_point')#,dtype=tf.float64)
```
Comment (hgandhi2411) [inline tensorflow_plugin/models/example-models/eds/real_build.py]: I think you can get rid of the commented out `#,dtype=tf.float64)` from all these lines if you are not using it.

**Comment by hgandhi2411** [comment]
I don't see why we need to have so many files in the eds folder. For example, there is a build_model.py, real_build.py and real_build_new.py, all for building the model. Similarly for running the model theer are duplicate files. Can we take out the unnecessary files?

**Comment by whitead** [comment]
I still see build and real_build

**Pull request merged**